plugins/follow_me/follow_me.h

```diff
@@ -0,0 +1,194 @@
+#pragma once
+
+#include <iostream>
+#include <cmath>
+
+
+namespace dronecore {
+
+class FollowMeImpl;
+
+class FollowMe
+{
+public:
+    explicit FollowMe(FollowMeImpl *impl);
+    ~FollowMe();
+
+    /**
+     * @brief Results of FollowMe operations
+     */
+    enum class Result {
+        SUCCESS = 0, /**< @brief %Request succeeded. */
+        NO_DEVICE, /**< @brief No device connected. */
+        CONNECTION_ERROR, /**< @brief %Connection error. */
+        BUSY, /**< @brief Vehicle busy. */
+        COMMAND_DENIED, /**< @brief Command denied. */
+        TIMEOUT, /**< @brief %Request timeout. */
+        UNKNOWN /**< @brief Unknown error. */
+    };
+
+    /**
+     * @brief FollowMe configuration.
+     * Reference: https://docs.px4.io/en/advanced_config/parameter_reference.html#follow-target
+     */
+    class Configuration
+    {
+    public:
+        /**
+         * @brief Side to follow target from
+         */
+        enum class FollowDirection {
+            NONE = -1,
+            FRONT_RIGHT = 0, /**< @brief Follow from front right. */
+            BEHIND = 1, /**< @brief Follow from behind. */
+            FRONT = 2, /**< @brief Follow from front. */
+            FRONT_LEFT = 3 /**< @brief Follow from front left. */
+        };
+
+        // Default configuration
+        ///////////////////////////////////////////////////////////
+        /**< @brief Default follow target height. */
+        static constexpr const float DEF_HEIGHT_ABOVE_HOME = 8.0f;
```

Comment:
I don't particularly like these constants in the header file. If there is some way to prevent that would be good.
Also, they all need to have units!

If you just need to have a default config, why don't you add default values to the Config. I mean like this:

```
struct Config {
   float min_altitude_m = 5.0f;
   FollowDirection follow_direction = FollowDirection::BEHIND;
};
```

Then, the default config can be created with:
```
Config config {};
```

Reply:
Constants are required for validating against values set by application. As it is header file, lets define them in `.cpp`